easybuild/tools/build_log.py

```diff
@@ -52,12 +53,29 @@
 DEPRECATED_DOC_URL = 'http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html'
 
 
-class EasyBuildError(Exception):
+# 1st argument (a LoggedException instance) is ignored
+def log_error_no_raise(_, logger, msg):
+    """Utility function to log an error with raising an exception."""
+    if hasattr(logger, 'raiseError'):
+        orig_raise_error = logger.raiseError
+        logger.raiseError = False
+
+    logger.error(msg)
+
+    if hasattr(logger, 'raiseError'):
+        logger.raiseError = orig_raise_error
+
+
+class EasyBuildError(LoggedException):
     """
     EasyBuildError is thrown when EasyBuild runs into something horribly wrong.
     """
-    def __init__(self, msg):
-        Exception.__init__(self, msg)
+    # make sure EasyBuildError isn't being raised again to avoid infinite recursion
```

Comment:
add comment that this is derecated and to be removed in `3.0` (then the default `error` method should be ok)